mobile/examples/object_detections/iOS/ORTObjectDetection/ModelHandler.swift

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AVFoundation
+import Accelerate
+import CoreImage
+import Darwin
+import Foundation
+import UIKit
+
+
+// Result struct
+struct Result {
+    let processTimeMs: Double
+    let inferences: [Inference]
+}
+
+// Inference struct for ssd model
+struct Inference {
+    let score: Float
+    let className: String
+    let rect: CGRect
+    let displayColor: UIColor
+}
+
+// Information about a model file or labels file.
+typealias FileInfo = (name: String, extension: String)
+
+enum OrtModelError : Error {
+    case error(_ message: String)
+}
+
+class ModelHandler: NSObject {
+    
+    // MARK: - Inference Properties
+    let threadCount: Int32
+    let threshold: Float = 0.5
+    let threadCountLimit = 10
+    
+    // MARK: - Model Parameters
+    let batchSize = 1
+    let inputChannels = 3
+    let inputWidth = 300
+    let inputHeight = 300
+    
+    private let colors = [
+        UIColor.red,
+        UIColor(displayP3Red: 90.0/255.0, green: 200.0/255.0, blue: 250.0/255.0, alpha: 1.0),
+        UIColor.green,
+        UIColor.orange,
+        UIColor.blue,
+        UIColor.purple,
+        UIColor.magenta,
+        UIColor.yellow,
+        UIColor.cyan,
+        UIColor.brown
+    ]
+    
+    private var labels: [String] = []
+    
+    init?(threadCount: Int32 = 1) {
```

Comment:
why `init?` instead of `init`?

Reply:
Originally I thought I can create a failable initializer for modelhandler, so that I can put all the loading models and label files (which can `return nil` therefore requires a failable init) and OrtSession initialization in it. But tried for several times it causes the app hung from execution. Still looking into it.